### PR TITLE
Missing string option fields

### PR DESCRIPTION
**Pull request by luissfx**

### CODE REVIEW

File: rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt [hunk 1]
```diff
@@ -174,10 +174,38 @@ public class NumberOptionBuilder(name: String, description: String) :
 public class StringChoiceBuilder(name: String, description: String) :
     BaseChoiceBuilder<String>(name, description, ApplicationCommandOptionType.String) {
 
+    private var _minLength: Optional<Int> = Optional.Missing()
+
+    /**
+     * The minimum string length (minimum of 0).
```
Comment (Lukellmann) [inline rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt:180]:
```suggestion
     * The minimum allowed string length (minimum of `0`, maximum of `6000`).
```
see https://github.com/discord/discord-api-docs/commit/5981d3fa3893a7f45fc7272d7a40796d13ad03b6

File: rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt [hunk 2]
```diff
@@ -174,10 +174,38 @@ public class NumberOptionBuilder(name: String, description: String) :
...
+     */
+    public var minLength: Int? by ::_minLength.delegate()
+
+    private var _maxLength: Optional<Int> = Optional.Missing()
+
+    /**
+     * The maximum string length (minimum of 1).
```
Comment (Lukellmann) [inline rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt:187]:
```suggestion
     * The maximum allowed string length (minimum of `1`, maximum of `6000`).
```
see https://github.com/discord/discord-api-docs/commit/5981d3fa3893a7f45fc7272d7a40796d13ad03b6

**Review by Lukellmann — COMMENTED** [review]
looks good overall, but please replace `Optional<Int>` with `OptionalInt`

**Comment by luissfx** [comment]
Done, thanks for the suggestions!

**Review by Lukellmann — APPROVED** [review]

**Pull request merged**